lib/luaT/luaT.c

```diff
@@ -497,27 +496,65 @@ int luaT_classmodulename(const char *tname, char *module_name)
   return tname[n] == '.';
 }
 
-/* Lua only functions */
+void traverse_tables(lua_State *L, const char *tname, int *L_idx)
```

Comment:
I think I prefer relative indices in this case as they are easier to read (-1 = what I just pushed on the stack). I'm not sure if there are any performance implications but I can't imagine luaT_newmetatable being on a critical path.